### PR TITLE
[JENKINS-33815] Fixing FolderComputation to properly handle AbortException

### PR DESCRIPTION
**Pull request by jglick**
Needed to do https://github.com/jenkinsci/github-branch-source-plugin/pull/35 properly.

@reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
:bee:

File: src/main/java/com/cloudbees/hudson/plugins/folder/computed/FolderComputation.java [hunk 1]
```diff
@@ -113,8 +112,12 @@ public void run() {
             listener.getLogger().println("Aborted");
             _result = Result.ABORTED;
         } catch (Exception x) {
-            // TODO skip stack trace for AbortException
-            x.printStackTrace(listener.fatalError("Failed to recompute children of " + getDisplayName()));
+            String message = "Failed to recompute children of " + folder.getFullDisplayName();
+            if (x instanceof AbortException) {
+                listener.fatalError(message + ": " + x.getMessage());
```
Comment (recena) [inline src/main/java/com/cloudbees/hudson/plugins/folder/computed/FolderComputation.java]: @jglick In this way your are **forcing** the message with *Failed to recompute children of*. I prefer something like that:

```
listener.fatalError("%n" + x.getMessage() + "%n%n");
```

Comment (recena) [inline src/main/java/com/cloudbees/hudson/plugins/folder/computed/FolderComputation.java]: @jglick Or, I would use a `LOGGER`...

```
String message = "Failed to recompute children of " + folder.getFullDisplayName();
LOGGER.log(Level.WARNING, message + ": " + x.getMessage(), x)
listener.fatalError(x.getMessage());
```

File: src/main/java/com/cloudbees/hudson/plugins/folder/computed/FolderComputation.java [hunk 2]
```diff
@@ -110,11 +109,16 @@ public void run() {
...
             listener.getLogger().println("Aborted");
             _result = Result.ABORTED;
         } catch (Exception x) {
-            // TODO skip stack trace for AbortException
-            x.printStackTrace(listener.fatalError("Failed to recompute children of " + getDisplayName()));
+            LOGGER.log(Level.FINE, "recomputation of " + folder.getFullName() + " failed", x);
+            if (x instanceof AbortException) {
+                listener.fatalError(x.getMessage());
+            } else {
+                x.printStackTrace(listener.fatalError("Failed to recompute children of " + folder.getFullDisplayName()));
```
Comment (recena) [inline src/main/java/com/cloudbees/hudson/plugins/folder/computed/FolderComputation.java]: @jglick Why are you using `folder.getFullDisplayName()` and `folder.getFullName()`? Both return the same?

**Comment by recena** [comment]
:bee:

File: src/main/java/com/cloudbees/hudson/plugins/folder/computed/FolderComputation.java (hunk 2 above)
Comment (jglick) [inline src/main/java/com/cloudbees/hudson/plugins/folder/computed/FolderComputation.java]: No, `fullName` would be for example `myorg/feature%2Fstuff` where `fullDisplayName` might be `My Org » feature/stuff`.

Comment (recena) [inline src/main/java/com/cloudbees/hudson/plugins/folder/computed/FolderComputation.java]: Thanks.

**Pull request merged**